suse_migration_services/units/kernel_reboot.py

```diff
@@ -0,0 +1,82 @@
+# Copyright (c) 2018 SUSE Linux LLC.  All rights reserved.
+#
+# This file is part of suse-migration-services.
+#
+# suse-migration-services is free software: you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# suse-migration-services is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with suse-migration-services. If not, see <http://www.gnu.org/licenses/>
+#
+
+# project
+from suse_migration_services.command import Command
+from suse_migration_services.defaults import Defaults
+
+from suse_migration_services.exceptions import (
+    DistMigrationKernelRebootException
+)
+
+
+def main():
+    """
+    DistMigration reboot with  new kernel
+
+    Reads the info of the new kernel and loads the latest (new migrated kernel)
+    version for rebooting.
+    """
+    target_kernel = Defaults.get_target_kernel()
+    target_initrd = Defaults.get_target_initrd()
+    try:
+        Command.run(
+            ['kexec', '--unload']
+        )
+        Command.run(
+            [
+                'kexec',
+                '--load', target_kernel,
+                '--initrd', target_initrd,
+                '--command-line', _get_cmdline(target_kernel)
+            ]
+        )
+        Command.run(
+            ['kexec', '--exec']
+        )
+    except Exception as issue:
+        Command.run(
+            ['kexec', '--unload']
+        )
+        raise DistMigrationKernelRebootException(
+            'Kernel rebooting failed with {0}'.format(
+                issue
+            )
+        )
+
+
+def _get_cmdline(vmlinuz):
+    grub_config_file = Defaults.get_grub_config_file()
+    pattern = '"linux\t\\K.+"'
+    try:
+        cmd_line = Command.run(
+            ['grep', '-oP', '-m', 1, pattern, grub_config_file]
+        )
```

Comment:
IMHO we should use Python to get the information from the grub config file rather than using grep.